### ISSUE TITLE
metrictank cant store points after falling behind primary

### ISSUE THREAD

**Issue by woodsaj** [raintank/metrictank#261]
If a metrictank node falls behind on processing for any reason, yet the primary node is still up to date, then the lagging node is unable to store points.

```
[aggmetric.go:433 Add()] [E] cant write to chunk that has already been saved.
```

This is happening because the lagging node is receiving the metricpersist messages before it has finished processing all data for the chunk.

non-primary nodes should always be able to write to chunks, regardless of whether they are marked as having been written to cassandra.

**Comment by woodsaj** [comment]
metrictank-1-production (new cluster) is really struggling and i think it is because every metric is resulting in an error log entry which is slowing things down.  This needs to be fixed fairly urgently.

**Comment by Dieterbe** [comment]
At first glance I don't see any saturation (e.g. iowait, cpu usage, network, ...) on mt-1-prod that seems to cause this.
Will look into this more, but I suspect it may be something with kafka

**Comment by woodsaj** [comment]
???? the problem is very clearly explained in the first post to the issue.

**Comment by Dieterbe** [comment]
yes i understand OP and agree it should be fixed, but my comment was to your 2nd post: i see nothing that would indicate the error logging is a cause of a problem. (rather just a sympton)

**Comment by Dieterbe** [comment]
so #263 does this. i haven't reproduced the scenario in dev yet, the quickest and still relatively safe way to test is probably just roll out to just mt-1 and keep it as non-primary.

it's still not a very clean solution though IMHO.  I wonder if we should just send metricpersist messages into the metrics topic. we would need to make sure we use the same keying/partitioning scheme so that the metricpersist messages go into the right metrics topic. but it would assure proper ordering.  maybe that could be a longer term solution.

**Comment by Dieterbe** [comment]
fixed by #263 
as for the mt-1-prod problems, that was because ES was responding slowly, and metrictank applied backpressure into the ingestion.
